test/unit/module/core/bit_cast.cpp

```diff
@@ -21,7 +21,33 @@ TTS_CASE_TPL("Check return types of bit_cast", eve::test::simd::all_types)
   TTS_EXPR_IS(eve::bit_cast(T(), eve::as<it_t>()), it_t);
   TTS_EXPR_IS(eve::bit_cast(T(), eve::as<ut_t>()), ut_t);
   TTS_EXPR_IS(eve::bit_cast(T(), eve::as<T>()), T);
-  TTS_EXPR_IS(eve::bit_cast(T(), eve::as<u8_t>()), u8_t);
+  // for casts between different sizes of scalar underlying types,
+  // we have no guarantee that underlying types for wide are the same.
+  if constexpr( sizeof(T) == sizeof(u8_t) ) TTS_EXPR_IS(eve::bit_cast(T(), eve::as<u8_t>()), u8_t);
+  if constexpr( sizeof(v_t) >= 2 )
+  {
+    using u16_t = eve::as_wide_t<std::int16_t, eve::fixed<eve::cardinal_v<T> * sizeof(v_t) / 2>>;
+    if constexpr( sizeof(T) == sizeof(u16_t) )
+      TTS_EXPR_IS(eve::bit_cast(T(), eve::as<u16_t>()), u16_t);
+  }
```

Comment:
make a using I guess.

```
template <typename T, typename Scalar>
using same_size_of = eve::wide<Scalar, 
     std::max(1, eve::fixed< T::size() * sizeof(eve::element_type_t<T>) / sizeof(Scalar))>>;
```

The the bunch of if constexprs go away and you test up scaling too.

Reply:
This will not work for RISC-V:
Suppose we have input `wide<int, 1>` and want to cast it to `wide` with `char`. For cardinal calculations we get `std::max(1, eve::fixed< T::size() * sizeof(eve::element_type_t<T>) / sizeof(Scalar))= std::max(1, 1* 4/1) = 4`.  In result we get `wide<std::int8_t, 4>`, for vlen=128 it will occupy 1/4 of physical register in intrinsics interface. However, the original `wide<int, 1>` will use 1/2 of physical register.

It seems this can be partially achieved by using `sizeof(wide<int, 1>)` instead of  `T::size() * sizeof(eve::element_type_t<T>)`, but we still have an error when we try to have `bit_cast` , for example, from `wide<std::int8_t, 4>` to any `wide` with `double`. `wide<std::int8_t, 4>` will occupy 1/4 of physical register, but any `wide` with double will use at least use 1 physical register, so `bit_cast` will be unavailable anyway.

I can remove additional type checks, and keep only `u8_t`, I think for this case we can check without `if constexpr` stuff.

What do you think?

Reply:
You still keep the if constexpr check for sizes matching. But you remove some

Reply:
Ok, got it. Seems done